packages/svelte/src/internal/client/dom/blocks/if.js

```diff
@@ -57,134 +50,115 @@ export function if_block(anchor_node, condition_fn, consequent_fn, alternate_fn)
-					}
-				}
-			} else if (hydrating) {
-				const comment_text = /** @type {Comment} */ (current_hydration_fragment?.[0])?.data;
-
-				if (
-					!comment_text ||
-					(comment_text === 'ssr:if:true' && !result) ||
-					(comment_text === 'ssr:if:false' && result)
-				) {
-					// Hydration mismatch: remove everything inside the anchor and start fresh.
-					// This could happen using when `{#if browser} .. {/if}` in SvelteKit.
-					remove(current_hydration_fragment);
-					set_current_hydration_fragment(null);
-					mismatch = true;
-				} else {
-					// Remove the ssr:if comment node or else it will confuse the subsequent hydration algorithm
-					current_hydration_fragment.shift();
-				}
+		if (condition === (condition = !!condition_fn())) return;
+
+		if (hydrating) {
+			const comment_text = /** @type {Comment} */ (current_hydration_fragment?.[0])?.data;
+
+			if (
+				!comment_text ||
+				(comment_text === 'ssr:if:true' && !condition) ||
+				(comment_text === 'ssr:if:false' && condition)
+			) {
+				// Hydration mismatch: remove everything inside the anchor and start fresh.
+				// This could happen using when `{#if browser} .. {/if}` in SvelteKit.
+				remove(current_hydration_fragment);
+				set_current_hydration_fragment(null);
+				mismatch = true;
+			} else {
+				// Remove the ssr:if comment node or else it will confuse the subsequent hydration algorithm
+				current_hydration_fragment.shift();
 			}
-
-			has_mounted = true;
 		}
 
-		// create these here so they have the correct parent/child relationship
-		consequent_effect ??= render_effect(
-			(/** @type {any} */ _, /** @type {import('#client').Effect | null} */ consequent_effect) => {
-				const result = block.v;
-
-				if (!result && consequent_dom !== null) {
-					remove(consequent_dom);
-					consequent_dom = null;
-				}
-
-				if (result && current_branch_effect !== consequent_effect) {
-					consequent_fn(anchor_node);
-					if (mismatch && current_branch_effect === null) {
-						// Set fragment so that Svelte continues to operate in hydration mode
-						set_current_hydration_fragment([]);
-					}
-					current_branch_effect = consequent_effect;
-					consequent_dom = block.d;
-				}
-
-				block.d = null;
-			},
-			block,
-			true
-		);
-		block.ce = consequent_effect;
-
-		alternate_effect ??= render_effect(
-			(/** @type {any} */ _, /** @type {import('#client').Effect | null} */ alternate_effect) => {
-				const result = block.v;
-
-				if (result && alternate_dom !== null) {
-					remove(alternate_dom);
-					alternate_dom = null;
-				}
-
-				if (!result && current_branch_effect !== alternate_effect) {
-					if (alternate_fn !== null) {
+		if (condition) {
+			if (consequent_effect) {
+				resume_effect(consequent_effect);
+			} else {
+				consequent_effect = render_effect(
+					() => {
+						consequent_fn(anchor_node);
+						consequent_dom = block.d;
+
+						if (mismatch) {
+							set_current_hydration_fragment([]);
```

Comment:
might be strange to understand else
```suggestion
							// Set fragment so that Svelte continues to operate in hydration mode
							set_current_hydration_fragment([]);
```